verifier/src/evidence.rs

```diff
@@ -32,18 +34,29 @@ impl<Q: AsRef<[u8]>> Evidence<Q> {
     pub fn new(quote: Quote3<Q>, collateral: Collateral) -> Result<Self> {
         // We perform any fallible conversions now to keep the verification focused on the values
         // and not the types/format.
+        let signed_qe_identity = SignedQeIdentity::try_from(collateral.qe_identity())?;
+        let qe_identity = QeIdentity::try_from(&signed_qe_identity)?;
```

Comment:
I'm not fond of how `Evidence` is contains fields that can be derived from other fields, but it still feels more ergonomic to have all the failures here in one place